### PR TITLE
Refactor P2P server.

### PR DESCRIPTION
**Pull request by jianjunz**
- Remove vendormodule as P2P server is open sourced. Developers may change authentication procedure by modifying source code.
- Remove unused code.
- Use modern JavaScript syntax.
- Add format and lint rules.

### CODE REVIEW

File: peerserver.js [hunk 1]
```diff
@@ -2,89 +2,73 @@
...
-var httpsOptions = {
+// Prepare for web server.
+const fs = require("fs");
+const path = require("path");
+const url = require('url');
+const config = require('./config');
+const forwardEventName = 'owt-message';
+
+const dirname = __dirname || path.dirname(fs.readlinkSync('/proc/self/exe'));
+const httpsOptions = {
```
Comment (taste1981) [inline peerserver.js]: Do we need to add a hard requirement of forcing TLS1.2 here?

File: peerserver.js [hunk 2]
```diff
@@ -2,89 +2,73 @@
...
 
-function authorization(socket, next){
-  var query=url.parse(socket.request.url,true).query;
-  var token=query.token;
-  var clientVersion=query.clientVersion;
-  var clientType=query.clientType;
+// Please modify this function if you need to add your own credential validation
+// procedure.
+function authenticate(socket, next){
+  const query=url.parse(socket.request.url,true).query;
```
Comment (taste1981) [inline peerserver.js]: format the code from line 52 to 58.

Comment (jianjunz) [reply peerserver.js]: Done.

File: peerserver.js (hunk 1 above)
Comment (jianjunz) [reply peerserver.js]: It depends on UA's behavior. @HuaChunbo may have suggestion on this?

**Pull request merged**